Simulator/Simulator/Simulator/python/plotting/network_activity/plot_fourieri_spike_activity.py

```diff
@@ -0,0 +1,120 @@
+import os
+import numpy as np
+from tqdm import tqdm
+
+from plot_network_spike_activity import get_args
+from plot_network_spike_activity import get_spikes_binaries_for_layer
+from plot_network_spike_activity import plot_data_and_lines
+from utils.utils_config import parse_yaml_config
+from utils.utils_simulator import parse_array_file
+from utils.utils_matplotlib import line_plot
+
+import matplotlib.pyplot as plt
+
+
+'''
+Almost the same code as plot_network_spike_activity.py but it will create a histogram-like of spike for each layer
+'''
+
+
+def get_fourier(signal):
+
+    fourier = np.fft.fft(signal)
+    freq = np.fft.fftfreq(len(signal)) * len(signal)
+    mask = freq > 0
+    fourier = 2.0 * np.abs(fourier / len(signal))
+
+    return fourier[mask], freq[mask]
+
+
+def make_data_for_layer_and_lines(binaries_for_layer, config):
+    '''
+    For each layer it will stack the spikes at each timestamps -> histogram effect
+    It will also create a set of lines that will split the layer points in the plot
+    And it will determine where on the y axis on the plot should the names of the layers be located
+
+    :param binaries_for_layer: list
+        output of @et_spikes_binaries_for_layer function
+
+    :param config: dict
+        the configuration
+
+    :return: 2 lists
+        @data_for_layer: list
+            each element keeps the @layer_name, the position o the @layer_name on the y axis, the x coordinates of
+            all the spikes of the layer, and the y coordinates for every spike in the layer
+        @lines: list
+            each element keeps 2 points, describing the ends of a line
+    '''
+    neuron_y_coord = 0
+
+    # I know that every neuron ran for the same amount of time
+    # so take the first one and see
+    array_file = os.path.join(config['layers_folder'], binaries_for_layer[0]['layer_name'], binaries_for_layer[0]['binaries'][0])
+    neuron_data = parse_array_file(array_file)
+    n_times = neuron_data['data'].shape[0]
+
+    data_for_layer = []
+    lines = [dict(x_coord=[0, n_times // 2], y_coord=[0, 0])]
+
+    for layer in tqdm(binaries_for_layer):
+        spike_histogram = np.zeros((n_times, ), dtype=np.int32)
+
+        for binary in layer['binaries']:
```

Comment:
make a function that gives you the histogram and uses it in the cumulative scipt

Reply:
Added function in the ```plot_network_cummulative_spike_activity.py``` and documented it